### DISCUSSION TITLE
Generate Clifford gates in matrix form

### DISCUSSION THREAD

**Discussion by ghost** [QuantumSavory/QuantumClifford.jl#103] [Q&A] [answered]
Hi Stefan:

I am thinking about generating all 24 single-qubit Clifford gates in matrix form. Is there a function to do that? Thanks

**Comment by Krastanov** [comment]
Yes, [`enumerate_cliffords`](https://krastanov.github.io/QuantumClifford.jl/dev/API/#QuantumClifford.enumerate_cliffords-Tuple{Any,%20Any}) and `enumerate_phases`.

```
julia> length(collect(enumerate_phases(enumerate_cliffords(1))))
24
```

**Comment by ghost** [comment]
Hi Stefan:

Yes this can generate all 24 single qubit clifford gates. But is there a simple way to get the matrix of each gate? 

Thanks

**Comment by Krastanov** [comment, marked as answer]
With the release of 0.8.12 today, this is easier.

- `stab_to_gf2` can be used to generate the binary matrix representing the gate
- `Operator` from `QuantumOptics.jl` can be used to convert `CliffordOperator` into a unitary matrix

If I understand correctly, you can do what you want by writing:

```julia-repl
julia> using QuantumClifford, QuantumOptics

julia> all_1q_clifford_unitaries = Operator.(enumerate_phases(enumerate_cliffords(1)));

julia> all_1q_clifford_unitaries[12]
Operator(dim=2x2)
  basis: Spin(1/2)
  0.5+0.0im  0.0-0.5im
 -0.5+0.0im  0.0-0.5im
```
